### PR TITLE
Oracle Connection Pooling implementation of issues/1655 

### PR DESCRIPTION
**Pull request by Moritz-Langer**
# Overview
This is an implementation of connection pooling in oracle. It leaves the decision up to the environment, if the variables 
`ORACLE_POOL_MIN` and `ORACLE_POOL_MAX` are found in the environment, a Pool is created with min and max connections according to the variable values. If they are not set, than single connections are used, just like before. 
This should lead to no breaks for current users and is easy to set up. 

Tests and Documentation are added as well.

# Related Issue / discussion
Related to https://github.com/geopython/pygeoapi/issues/1655  

# Additional information

# Dependency policy (RFC2)

- [x] I have ensured that this PR meets [RFC2](https://pygeoapi.io/development/rfc/2) requirements

# Updates to public demo

- [ ] I have ensured that breaking changes to the [pygeoapi master demo server](https://github.com/geopython/demo.pygeoapi.io) have been addressed
  - [ ] https://github.com/geopython/demo.pygeoapi.io/blob/master/services/pygeoapi_master/local.config.yml

# Contributions and licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution
- [x] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

**Comment by Moritz-Langer** [comment]
@8luewater fyi

**Review by totycro — APPROVED** [review]

File: pygeoapi/provider/oracle.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # =================================================================
 #
 # Authors: Andreas Kosubek <andreas.kosubek@ama.gv.at>
+# Authors: Moritz Langer <moritz.b.langer@gmail.com>
 #
 # Copyright (c) 2023 Andreas Kosubek
```
Comment (tomkralidis) [inline pygeoapi/provider/oracle.py:6]: Can you also add yourself to the copyright?

File: docs/source/data-publishing/ogcapi-features.rst [hunk 2]
```diff
@@ -387,6 +387,19 @@ Extra properties
 Extra properties is a list of strings which are added as fields for data retrieval in the SELECT clauses. They
 can be used to return expressions computed by the database.
 
+Session Pooling
```
Comment (tomkralidis) [inline docs/source/data-publishing/ogcapi-features.rst:390]: Suggested formatting updates:

```
Session Pooling
""""""""""""""""

Configured using environment variables.

.. code-block:: bash

   export ORACLE_POOL_MIN=2
   export ORACLE_POOL_MAX=10


The ``ORACLE_POOL_MIN`` and ``ORACLE_POOL_MAX`` environment variables are used to trigger session pool creation in the Oracle Provider and the ``DatabaseConnection`` class. See https://python-oracledb.readthedocs.io/en/latest/api_manual/module.html#oracledb.create_pool for documentation of the ``create_pool`` function.

If none or only one of the environment variables is set, session pooling will not be activated and standalone connections are established at every request.
```

File: pygeoapi/provider/oracle.py [hunk 3]
```diff
@@ -33,6 +34,8 @@
 import oracledb
```
Comment (tomkralidis) [inline pygeoapi/provider/oracle.py:34]: Can we update the import ordering as follows:

```python
import importlib
import json
import logging
import os
import threading
from typing import Optional

import oracledb
import pyproj

from pygeoapi.api import DEFAULT_STORAGE_CRS
```

**Review by tomkralidis — APPROVED** [review]

**Review by tomkralidis — CHANGES_REQUESTED** [review]
Requesting a few minor changes -- good work!

**Comment by Moritz-Langer** [comment]
Hi Tom,
thank you for the feedback and the review. I have adjusted according to your comments.

File: pygeoapi/provider/oracle.py (hunk 3 above)
Comment (Moritz-Langer) [reply pygeoapi/provider/oracle.py:38]: Changed accordingly

File: docs/source/data-publishing/ogcapi-features.rst (hunk 2 above)
Comment (Moritz-Langer) [reply docs/source/data-publishing/ogcapi-features.rst:390]: Changed accordingly

File: pygeoapi/provider/oracle.py (hunk 1 above)
Comment (Moritz-Langer) [reply pygeoapi/provider/oracle.py:6]: Changed accordingly

**Review by tomkralidis — APPROVED** [review]

**Pull request merged**